Test suite does not compile with Clang 13.1

```
$ clang --version
Apple clang version 13.1.6 (clang-1316.0.21.2.5)
Target: arm64-apple-darwin21.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
$ make -f xca6ll.gmk VARIETY=cool testrun
zmess.c:151:7: error: variable 'found' set but not used [-Werror,-Wunused-but-set-variable]
  int found = 0;
      ^
1 error generated.
```

--- a/code/zmess.c
+++ b/code/zmess.c
@@ -148,7 +148,6 @@ static int state[myrootCOUNT];
  */
 static void report(mps_arena_t arena, const char *pm, Bool discard)
 {
-  int found = 0;
   char mFound = '\0';
   mps_message_type_t type;
 
@@ -160,7 +159,6 @@ static void report(mps_arena_t arena, const char *pm, Bool discard)
 
     cdie(mps_message_get(&message, arena, type),
          "get");
-    found += 1;
 
     switch(type) {
       case mps_message_type_gc_start(): {

